source/merge-deep.d.ts

```diff
@@ -63,6 +63,11 @@ type MergeDeepRecord<
 > = DoMergeDeepRecord<OmitIndexSignature<Destination>, OmitIndexSignature<Source>, Options>
 & Merge<PickIndexSignature<Destination>, PickIndexSignature<Source>>;
 
+/**
+The maximal depth to use when looking for rest types in {@link PickRestType}
+*/
+type PickRestTypeMaxDepth = 50;
```

Comment:
It would be good with a good comment about why it's 50, even if the reason is that it was just randomly picked as something sensible.

Reply:
Added:

> 50 is arbitrary selected through manual testing. 100 was too much.